bids/layout/layout.py

```diff
@@ -335,6 +343,8 @@ def get(self, return_type='object', target=None, extensions=None,
 
         if derivatives == True:
             derivatives = list(self.derivatives.keys())
+        if derivatives:
```

Comment:
Since `derivatives == True` always produces a list, no need to double-hit it:

```suggestion
        elif derivatives:
```